Mathlib/Analysis/Complex/AbelLimit.lean

```diff
@@ -0,0 +1,253 @@
+/-
+Copyright (c) 2024 Jeremy Tan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeremy Tan
+-/
+import Mathlib.Analysis.Complex.Basic
+import Mathlib.Analysis.SpecificLimits.Normed
+
+/-!
+# Abel's limit theorem
+
+If a real or complex power series for a function has radius of convergence 1 and the series is only
+known to converge conditionally at 1, Abel's limit theorem gives the value at 1 as the limit of the
+function at 1 from the left. "Left" for complex numbers means "within a Stolz set", a subset of the
+open unit disc with 1 on its boundary.
+
+## Main theorems
+
+* `Complex.tendsto_tsum_power_nhdsWithin_stolzSet`: Abel's limit theorem for complex power series.
+* `Real.tendsto_tsum_power_nhdsWithin_lt`: Abel's limit theorem for real power series.
+
+## References
+
+* https://planetmath.org/proofofabelslimittheorem
+* https://en.wikipedia.org/wiki/Abel%27s_theorem
+-/
+
+
+open Filter Finset
+
+open scoped BigOperators Topology
+
+namespace Complex
+
+section StolzSet
+
+/-- The Stolz set for a given `M`, roughly teardrop-shaped with the tip at 1 but tending to the
+open unit disc as `M` tends to infinity. -/
+def stolzSet (M : ℝ) : Set ℂ := {z | ‖z‖ < 1 ∧ ‖1 - z‖ < M * (1 - ‖z‖)}
+
+variable {M : ℝ}
+
+theorem stolzSet_empty (hM : M ≤ 1) : stolzSet M = ∅ := by
+  ext z
+  rw [stolzSet, Set.mem_setOf, Set.mem_empty_iff_false, iff_false, not_and, not_lt, ← sub_pos]
+  intro zn
+  calc
+    _ ≤ 1 * (1 - ‖z‖) := mul_le_mul_of_nonneg_right hM zn.le
+    _ = ‖(1 : ℂ)‖ - ‖z‖ := by rw [one_mul, norm_one]
+    _ ≤ _ := norm_sub_norm_le _ _
+
+/-- The intersection of `stolzSet M` with the real axis is `((1 - M) / (1 + M), 1)` for `1 < M`. -/
+theorem preimage_stolzSet_eq_Ioo (hM : 1 < M) :
```

Comment:
I don't think this is needed.

Reply:
My idea was to use this to show `nhdsWithin_lt_le_nhdsWithin_stolzSet`. Will that approach work and if so will it be shorter than your golf?

Reply:
You don't need the precise statement on the intersection (note that I only used that the open interval from `0` to `1` is contained in any nonempty Stolz set in my suggested proof), so I doubt that using it will give you a shorter (in total) proof.